alphacsc/init_dict.py

```diff
@@ -18,45 +18,254 @@
 from .utils.dictionary import get_uv, get_D
 
+
 ried = custom_distances.roll_invariant_euclidean_distances
 tied = custom_distances.translation_invariant_euclidean_distances
 
 
+def get_init_strategy(n_times_atom, shape, random_state, D_init,
+                      D_init_params):
+    """Returns dictionary initialization strategy.
+
+    Parameters
+    ----------
+    n_times_atom : int
+        The support of the atom.
+    shape: tuple
+        Expected shape of the dictionary. (n_atoms, n_channels + n_times_atoms)
+    or (n_atoms, n_channels, n_times_atom)
+    random_state: int or np.random.RandomState
+        A seed to generate a RandomState instance or the instance itself.
+    D_init : str or array, shape (n_atoms, n_channels + n_times_atoms) or \
+                           shape (n_atoms, n_channels, n_times_atom)
+        The initial atoms or an initialization scheme in
+        {'kmeans' | 'ssa' | 'chunk' | 'random' | 'greedy'}.
+    D_init_params : dict
+        Dictionnary of parameters for the kmeans init method.
+    """
+    if isinstance(D_init, np.ndarray):
+        return IdentityStrategy(shape, D_init)
+    elif D_init is None or D_init == 'random':
+        return RandomStrategy(shape, random_state)
+    elif D_init == 'chunk':
+        return ChunkStrategy(n_times_atom, shape, random_state)
+    elif D_init == "kmeans":
+        return KMeansStrategy(n_times_atom, shape, random_state, D_init_params)
+    elif D_init == 'ssa':
+        return SSAStrategy(n_times_atom, shape, random_state)
+    elif D_init == 'greedy':
+        return GreedyStrategy(shape, random_state)
+    else:
+        raise NotImplementedError('It is not possible to initialize uv'
+                                  ' with parameter {}.'.format(D_init))
+
+
+class IdentityStrategy():
+    """A class that creates a dictionary from a specified array.
+
+    Parameters
+    ----------
+    shape: tuple
+        Expected shape of the dictionary. (n_atoms, n_channels + n_times_atoms)
+    or (n_atoms, n_channels, n_times_atom)
+    D_init : array of shape (n_atoms, n_channels + n_times_atoms) or \
+                      shape (n_atoms, n_channels, n_times_atom)
+    """
+
+    def __init__(self, shape, D_init):
+
+        assert shape == D_init.shape
+        self.D_init = D_init
+
+    def initialize(self, X):
+        return self.D_init.copy()
+
+
+class RandomStrategy():
+    """A class that creates a random dictionary for a specified shape.
+
+    Parameters
+    ----------
+    shape: tuple
+        Expected shape of the dictionary. (n_atoms, n_channels + n_times_atoms)
+    or (n_atoms, n_channels, n_times_atom)
+    random_state: int or np.random.RandomState
+        A seed to generate a RandomState instance or the instance itself.
+    """
+
+    def __init__(self, shape, random_state):
+
+        self.shape = shape
+        self.rng = check_random_state(random_state)
+
+    def initialize(self, X):
+        return self.rng.randn(*self.shape)
```

Comment:
Ok. I've updated. I thought it would be good to have different dictionaries each time the function is called.

Reply:
This can be achieve by passing a `rng` as `random_state` in the init.
But in general, I find it less surprising like this, `random_state` fixes the output for `initialize`.